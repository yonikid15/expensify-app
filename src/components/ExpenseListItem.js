import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        {
            (id && description && amount) ? (
                <div>
                    <Link to={`/edit/${id}`}>
                        <h3>{description}</h3>
                    </Link>
                    <p>{amount} - {createdAt}</p>
                </div>
            ) : (
                <p>Expense Error</p>
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default ExpenseListItem;