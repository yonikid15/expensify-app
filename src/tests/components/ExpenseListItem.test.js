import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with correct expense', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseListItem with invalid expense', () => {
    const wrapper = shallow(<ExpenseListItem {...{}} />);
    expect(wrapper).toMatchSnapshot();
});