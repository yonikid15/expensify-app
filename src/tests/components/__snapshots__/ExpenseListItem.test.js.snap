// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with correct expense 1`] = `
<div>
  <div>
    <Link
      replace={false}
      to="/edit/1"
    >
      <h3>
        Gum
      </h3>
    </Link>
    <p>
      195
       - 
      0
    </p>
  </div>
</div>
`;

exports[`should render ExpenseListItem with invalid expense 1`] = `
<div>
  <p>
    Expense Error
  </p>
</div>
`;
